libs/wingsdk/src/target-tf-aws/api.ts

```diff
@@ -390,70 +390,105 @@ class WingRestApi extends Construct {
+    };
+
+    // let OAS30 = CORS
+    //   ? {
+    //       ...OAS30_BASE,
+    //       "x-amazon-apigateway-cors": {
+    //         allowMethods: ["GET", "OPTIONS", "POST"],
+    //         allowHeaders: ["*"],
+    //         maxAge: 0,
+    //         allowCredentials: false,
+    //         allowOrigins: ["*"],
+    //       },
+    //     }
+    //   : OAS30_BASE;
+
+    const apigwCorsConfig = props.cors
+      ? {
+          corsConfiguration: {
+            // renamed allowOrigin to allowOrigins
+            allowOrigins: props.cors.allowOrigins
+              ? props.cors.allowOrigins
+              : ["*"],
+            allowMethods: props.cors.allowMethods
+              ? props.cors.allowMethods
+              : ["GET", "POST", "PUT", "DELETE", "HEAD", "OPTIONS"],
+            allowHeaders: props.cors.allowHeaders
+              ? props.cors.allowHeaders
+              : ["Content-Type", "Authorization", "X-Requested-With"],
+            exposeHeaders: props.cors.exposeHeaders
+              ? props.cors.exposeHeaders
+              : [],
+            allowCredentials: props.cors.allowCredentials
+              ? props.cors.allowCredentials
+              : false,
+            // added maxAge
+            maxAge: props.cors.maxAge ? props.cors.maxAge : 0,
+          },
+        }
+      : {};
 
-    this.api = new ApiGatewayRestApi(this, "api", {
+    this.api = new Apigatewayv2Api(this, "api", {
       name: ResourceNames.generateName(this, NAME_OPTS),
-      // Lazy generation of the api spec because routes can be added after the API is created
+      protocolType: "HTTP",
       body: Lazy.stringValue({
         produce: () => {
-          const injectGreedy404Handler = (openApiSpec: OpenApiSpec) => {
-            openApiSpec.paths = {
-              ...openApiSpec.paths,
-              ...defaultResponse,
-            };
-            return openApiSpec;
-          };
-          return JSON.stringify(injectGreedy404Handler(props.apiSpec));
+          props.apiSpec = "random assignment so compiler doesn't complain";
+          return JSON.stringify(OAS30);
         },
       }),
+      ...apigwCorsConfig,
     });
 
-    this.deployment = new ApiGatewayDeployment(this, "deployment", {
-      restApiId: this.api.id,
-      lifecycle: {
-        createBeforeDestroy: true,
-      },
-      triggers: {
-        // Trigger redeployment when the api spec changes
-        redeployment: Fn.sha256(this.api.body),
-      },
-    });
-
-    this.stage = new ApiGatewayStage(this, "stage", {
-      restApiId: this.api.id,
-      stageName: STAGE_NAME,
-      deploymentId: this.deployment.id,
+    this.stage = new Apigatewayv2Stage(this, "stage", {
+      apiId: this.api.id,
+      name: "$default",
+      autoDeploy: true,
     });
 
-    //should be exported from here, otherwise won't be mapped to the right token
-    this.url = this.stage.invokeUrl;
+    this.url = `https://${this.api.id}.execute-api.${this.region}.amazonaws.com`;
```

Comment:
why not https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apigatewayv2_api#api_endpoint ?